source/Calamari.Common/Features/StructuredVariables/YamlFormatVariableReplacer.cs

```diff
@@ -155,7 +155,7 @@ void LogReplacement(string key)
             }
             catch (Exception e) when (e is SyntaxErrorException || e is SemanticErrorException)
             {
-                throw new StructuredConfigFileParseException(e.Message, e);
+                throw new StructuredConfigFileParseException(e.ToString(), e);
```

Comment:
The exception message no longer includes the Start and End locations covered in the `ParsingErrorPositionIsReportedToUser` test. ToString includes these and has the same string formatting as 8.2.1.